remotespark/livyclientlib/livysession.py

```diff
@@ -90,11 +89,14 @@ def create_sql_context(self):
         self.ipython_display.writeln("Creating HiveContext as 'sqlContext'")
         command = self._get_sql_context_creation_command()
         try:
-            command.execute(self)
+            (success, out) = command.execute(self)
         except LivyClientTimeoutError:
             raise LivyClientTimeoutError("Failed to create the SqlContext in time. Timed out after {} seconds."
                                          .format(self._wait_for_idle_timeout_seconds))
-        self.created_sql_context = True
+        if success:
```

Comment:
This feedback is not addressed on the latest iteration. We meant that you should at least add a unit test to the test_sparkcontroller.py file for when session.start() fails as well as a unit test to the test_kernelmagics.py file for when sparkcontroller.add_session() fails.